src/scanner-engine.ts

```diff
@@ -136,11 +136,17 @@ export function runScannerEngine(
 
   return new Promise<void>((resolve, reject) => {
     child.on('exit', code => {
-      if (code === 0) {
-        log(LogLevel.DEBUG, 'Scanner engine finished successfully');
-        resolve();
+      if (typeof code === 'number') {
+        if (code === 0) {
+          log(LogLevel.DEBUG, 'Scanner engine finished successfully');
+          resolve();
+        } else {
+          reject(new Error(`Scanner engine failed with code ${code}`));
+          process.exitCode = code;
```

Comment:
So, if I understand correctly, this change will make the bin/js APIs fail with exit code whenever the program exits? Do we want this for the JS API?

I'm wondering whether this could be confusing for users or not 🤔 No strong opinion on this, I am pointing this out to clarify intent

Reply:
> So, if I understand correctly, this change will make the bin/js APIs fail with exit code whenever the program exits?

exactly, I tested locally and the code is properly bubbled to the parent process when it is terminated. 

I wanted to be consistent in how we work with exit codes (setting the code instead of force exit). 

> Do we want this for the JS API?
I'm not sure I follow, I guess you mean do we want this in the sonar-scanner executable for npm; I say yes it is acceptable

Reply:
he means if someone uses the npm programmatically, we will change the exit code of the user's script. I'm not convinced about this to be honest. As a cli tool, absolutely makes sense. But a script developer could handle the exception and provide a workaround, so changing the exit code to error is not the expected behavior for an npm library IMHO.

that's a really good catch BTW

Reply:
Thanks for calling this out, I was able to simplify / rollback some changes in the latest commit
https://github.com/SonarSource/sonar-scanner-npm/pull/153/commits/d60a005cb606ee541e448033347adfb2797b540d